### ISSUE TITLE
Stream subscription events

### ISSUE THREAD

**Issue by schickling** [prisma/prisma1#585]
Additionally to invoking functions via subscription events, it would be great if subscription events could be streamed directly to different event targets such as AWS Kinesis, Kafka etc.

Until this feature is directly supported via Graphcool, an easy workaround is to use functions and forward events programatically.

**Comment by sorenbs** [comment]
We have decided to remove the top-level `functions` field. See #591

If sendConfirmEmail is a subscription that should put an item on a kinesis queue it should look like this:

```yml
subscriptions:
  sendConfirmEmail:
    handler:
      kinesis:
        stream: my-kinesis-stream
    query: ./src/sendConfirmEmail.graphql
```

**Comment by marktani** [comment]
This issue has been moved to [graphcool/graphcool-framework](https://github.com/graphcool/graphcool-framework/).

**Comment by schickling** [comment]
This is still highly relevant for Prisma. Reopening.

**Comment by akoenig** [comment]
@schickling Definitely! Would be great to push Prisma in the "Change Data Capture" space as well. I could imagine a [Debezium](http://debezium.io/)-like scenario. Prisms ships with Server-side subscriptions which already handles the first part of the problem: the possibility of pushing change events to a message broker. Imagine a scenario where you want to replicate all change events. In this case you also need a mechanism for creating a snapshot of the current state and emit those objects as "create" events. When this snapshot has been rolled out, the Server-side subscription(s) could kick in and stream the recent change events.

The Debezium model might be a good fit here. See this [part](http://debezium.io/docs/connectors/mysql/#how-the-mysql-connector-works) of the documentation for deeper insights how the actual MySQL connector works.

**Comment by schickling** [comment]
In relation to the Kafka integration, it would be interesting to evaluate whether [Apache Avro](https://avro.apache.org/docs/current/idl.html) should also be supported.

**Comment by rwatts3** [comment]
Yes, @akoenig I am running into this issue currently.
Rewriting an app that I wrote in Meteor, which featured a chat like interface with realtime messages and notifications.

I looked into rethinkdb briefly before coming over to prisma. 

Now that I am in the process of refactoring that application using Prisma, and Vue, I'm finding that the subscriptions system does not maintain existing data as you've mentioned above.
